libraries/protocol/include/graphene/protocol/fee_schedule.hpp

```diff
@@ -37,116 +37,74 @@ namespace graphene { namespace protocol {
    template<typename Operation>
    class fee_helper {
      public:
-      const typename Operation::fee_parameters_type& cget(const fee_parameters::flat_set_type& parameters)const
+      const typename Operation::fee_parameters_type& get(const fee_parameters::flat_set_type& parameters)const
       {
          auto itr = parameters.find( typename Operation::fee_parameters_type() );
-         FC_ASSERT( itr != parameters.end() );
-         return itr->template get<typename Operation::fee_parameters_type>();
-      }
-   };
+         if( itr != parameters.end() )
+            return itr->template get<typename Operation::fee_parameters_type>();
 
-   template<>
-   class fee_helper<account_create_operation> {
-     public:
-      const account_create_operation::fee_parameters_type& cget(const fee_parameters::flat_set_type& parameters)const
-      {
-         auto itr = parameters.find( account_create_operation::fee_parameters_type() );
-         FC_ASSERT( itr != parameters.end() );
-         return itr->get<account_create_operation::fee_parameters_type>();
+         static typename Operation::fee_parameters_type dummy;
+         return dummy;
       }
-      typename account_create_operation::fee_parameters_type& get(fee_parameters::flat_set_type& parameters)const
+      typename Operation::fee_parameters_type& mutable_get(fee_parameters::flat_set_type& parameters)const
       {
-         auto itr = parameters.find( account_create_operation::fee_parameters_type() );
-         FC_ASSERT( itr != parameters.end() );
-         return itr->get<account_create_operation::fee_parameters_type>();
+         auto itr = parameters.find( typename Operation::fee_parameters_type() );
+         if( itr != parameters.end() )
+            return itr->template get<typename Operation::fee_parameters_type >();
+         
+         static typename Operation::fee_parameters_type dummy = get(parameters);
+         parameters.insert(dummy);
+         return dummy;
```

Comment:
This seems wrong. As I understood, the `dummy` object didn't get moved to the `flat_set`, so returning `dummy` here is not returning a reference to the fee param object in the fee schedule, so the result of a later update would be not as expected.

Reply:
Will fix in another branch, not part of this branch any longer.